### ISSUE TITLE
Food upload functionality is broken

### ISSUE THREAD

**Issue by savchenko** [davidhealey/waistline#778]
Would be great to receive this notification _before_ all information is entered:

![image](https://github.com/davidhealey/waistline/assets/300146/867214c2-d3d0-42aa-a003-16274e6968a9)

**Comment by davidhealey** [comment]
But that isn't shown until you press the upload button is it?

**Comment by davidhealey** [comment]
I don't think we can check the end point without sending some data to it. Also between the check and you pressing the upload button the problem might occur.

Saving the data locally is probably the better option here, with the ability to upload it at a later date.

**Comment by fs-99** [comment]
local save would be huge, I got the same error today

**Comment by Baggypants** [comment]
I think this is related to to country setting. Unsetting it let me upload food from waistline

https://github.com/davidhealey/waistline/issues/895#issuecomment-2621430844
